include/aws/io/private/pkcs11_private.h

```diff
@@ -7,13 +7,61 @@
  */
 #include <aws/io/pkcs11.h>
 
+struct aws_string;
+
 /**
  * pkcs11_private.h
  * This file declares symbols that are private to aws-c-io but need to be
  * accessed from multiple .c files.
  */
 
+/* aws_pkcs11_t: equivalent to CK_SLOT_ID / CK_SESSION_HANDLE / CK_OBJECT_HANDLE.
+ * including PKCS#11 headers is weird, so outside of pkcs11.c we'll just use this typedef instead */
+typedef unsigned long aws_pkcs11_t;
```

Comment:
Is it better to include a hint on what type it is, "aws_pkcs11_t" does not tell me anything about it.

Reply:
I flip-flopped a lot on the best way forward here. I just replaced it with `unsigned long` in the "private API".

I tried it every way: exposing the actual PKCS#11 headers, unique typedefs for `aws_pkcs11_session_handle_t`, `aws_pkcs11_slot_id_t`, and just using `unsigned long`.

`unsigned long` felt the least confusing